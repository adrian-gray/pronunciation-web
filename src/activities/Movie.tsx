import React from "react";

import YouTube from "../components/YouTube";
import SplitHilite from "./../components/SplitHilite";

export default (props: {
  ipa: string;
  title: string;
  src: string;
  isUserAuth: boolean;
}) => {
  const { ipa, title, src } = props;

  return (
    <div className="head-space">
      <h3>
        {`English Pronunciation movie for `}
        <SplitHilite str={ipa} />
      </h3>

      <YouTube src={src} title={title} />
    </div>
  );
};
