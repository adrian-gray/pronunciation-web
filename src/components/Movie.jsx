import React from "react";

import SplitHilite from "./SplitHilite";

import { Typography, withStyles } from "@material-ui/core";

const styles = theme => ({
  headspace: theme.headspace,
  fullWidth: theme.fullWidth
});

const Movie = props => {
  const { classes, ipa, url } = props;

  return (
    <div className={classes.headspace}>
      <Typography variant="title" gutterBottom>
        {`English Pronuniciation movie for `}
        <SplitHilite str={ipa} />
      </Typography>

      <img className={classes.fullWidth} src={url} alt={`Pronouncing ${ipa}`} />
    </div>
  );
};

export default withStyles(styles)(Movie);
