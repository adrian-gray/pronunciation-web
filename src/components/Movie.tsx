import React from "react";

import SplitHilite from "./SplitHilite";

export default props => {
  const { ipa, url } = props;

  return (
    <div className="head-space">
      <h3>
        {`English Pronuniciation movie for `}
        <SplitHilite str={ipa} />
      </h3>

      <img className="full-width" src={url} alt={`Pronouncing ${ipa}`} />
    </div>
  );
};
