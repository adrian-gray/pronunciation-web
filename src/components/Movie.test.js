import React from "react";
import { shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import Movie from "./Movie";

afterEach(cleanup);

test("Movie", async () => {
  const props = {};

  const container = shallow(<Movie {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
