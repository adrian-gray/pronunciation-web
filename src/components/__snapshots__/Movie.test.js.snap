// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie 1`] = `undefined`;
